compiler/rustc_abi/src/lib.rs

```diff
@@ -1505,7 +1505,11 @@ impl BackendRepr {
 #[cfg_attr(feature = "nightly", derive(HashStable_Generic))]
 pub enum Variants<FieldIdx: Idx, VariantIdx: Idx> {
     /// Single enum variants, structs/tuples, unions, and all non-ADTs.
-    Single { index: VariantIdx },
+    Single {
+        /// Always 0 for non-enums/generators.
+        /// For enums without a variant, this is an invalid index!
```

Comment:
nit: "enums without a variant" is the same as "uninhabited enums" right? Might be better to use consistent terminology here?

I see `0-variant` is used later in this commit so could also use that name.

Reply:
No that's not the same. `Result<!, !>` is a 2-variant uninhabited enum.

Reply:
Got it, thank you!